build_tools/build_detect_platform

```diff
@@ -317,9 +317,8 @@ EOF
     # Test whether jemalloc is available
     if echo 'int main() {}' | $CXX $CFLAGS -x c++ - -o /dev/null -ljemalloc \
       2>/dev/null; then
-        PLATFORM_LDFLAGS="$PLATFORM_LDFLAGS -ljemalloc"
-        JAVA_LDFLAGS="$JAVA_LDFLAGS -ljemalloc"
         JEMALLOC=1
+        WITH_JEMALLOC_FLAG=1
```

Comment:
You are right. I should add more comments and explain that JEMALLOC could be enabled by either providing the library to the linker or using the flag and WITH_JEMALLOC_FLAG indicates the latter.